### PR TITLE
W/garden

### PR DESCRIPTION
**Pull request by willkunz13**
Merged in the current master already.   Could be reviewed further to ensure apiplants and plants are functioning properly.

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -16,4 +16,18 @@
   post '/profile', to: 'profiles#create'
 
   get '/plants', to: 'plants#index'
+
+	namespace :user do
```
Comment (nkeller1) [inline config/routes.rb:20]: I like the namespace of these routes!

File: db/migrate/20200412211316_create_garden_plants.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+class CreateGardenPlants < ActiveRecord::Migration[5.2]
+  def change
+    create_table :garden_plants do |t|
```
Comment (nkeller1) [inline db/migrate/20200412211316_create_garden_plants.rb:3]: Don't we need a date stamp or something here? Or is the plan to do this in a separate migration later?

File: spec/features/users/mygarden/garden_plants_spec.rb [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+      expect(current_path).to eq("/user/mygardens/#{@garden.id}")
+      expect(page).to have_content('plants in this garden. Find something you would like to grow and add them to keep track of what you have')
+		end
+
+		it "Add plant", :vcr do
+			visit '/'
+			fill_in "search", with: "tomato"
+			click_on "Search"
+			click_on "Tomato"
+			click_on "Add to MyGarden"
```
Comment (nkeller1) [inline spec/features/users/mygarden/garden_plants_spec.rb:40]: DO we want to add a flash message here?

File: spec/models/api_plant_spec.rb [hunk 4]
```diff
@@ -0,0 +1,32 @@
+require 'rails_helper'
+
+RSpec.describe ApiPlant, type: :model do 
+  
+  it "attributes" do
+
+    plant = ApiPlant.new(name: 'tomato',
```
Comment (nkeller1) [inline spec/models/api_plant_spec.rb:7]: I like having two separate plants. Good call!

File: spec/models/plant_spec.rb [hunk 5]
```diff
@@ -1,32 +1,9 @@
...
-          spread: 60,
-          sowing: "Direct seed indoors, transplant seedlings outside after hardening off",
-          spacing: 75)
-  end 
-    
-
-end
\ No newline at end of file
+describe Plant, type: :model do
+#        describe "validations" do
```
Comment (nkeller1) [inline spec/models/plant_spec.rb:2]: Do you want this code commented out for the time being or should it be removed?

File: app/views/user/mygardens/plants/index.html.erb [hunk 6]
```diff
@@ -0,0 +1,8 @@
+<h1> <%= @garden.name %> </h1>
+<% if @garden.plants.empty? %>
+	<h3> No plants in this garden.  Find something you would like to grow and add them to keep track of what you have planted </h3>
+<% else %>
+	<% @garden.plants.each do |plant| %>
```
Comment (nkeller1) [inline app/views/user/mygardens/plants/index.html.erb:5]: do we want to add the image here as well?

File: app/views/user/mygardens/_form.html.erb [hunk 7]
```diff
@@ -0,0 +1,19 @@
+<%= form_tag path, method: method do %>
```
Comment (nkeller1) [inline app/views/user/mygardens/_form.html.erb:1]: glad this works!

File: app/views/search/show.html.erb [hunk 8]
```diff
@@ -11,6 +11,21 @@
   <p>How to Sow: <%=  @plant.sowing %> </p>
   <p>Average Spacing <%=  @plant.spacing %> </p>
 </section>
+<% if current_user %>
+	<%= form_tag user_plants_path() do %>
```
Comment (nkeller1) [inline app/views/search/show.html.erb:15]: glad you got this working!

File: app/controllers/user/base_controller.rb [hunk 9]
```diff
@@ -0,0 +1,7 @@
+class User::BaseController < ApplicationController
+  before_action :require_user
+
+  def require_user
```
Comment (nkeller1) [inline app/controllers/user/base_controller.rb:4]: nice catch

File: db/migrate/20200412211316_create_garden_plants.rb (hunk 2 above)
Comment (willkunz13) [reply db/migrate/20200412211316_create_garden_plants.rb:3]: We could.  are you thinking for the calendar generation?

File: spec/models/plant_spec.rb (hunk 5 above)
Comment (willkunz13) [reply spec/models/plant_spec.rb:2]: probably still commented out.  it should probably have some required things, although then again users aren't creating it directly

File: app/views/user/mygardens/plants/index.html.erb (hunk 6 above)
Comment (willkunz13) [reply app/views/user/mygardens/plants/index.html.erb:5]: probably a future iteration.

File: app/views/search/show.html.erb (hunk 8 above)
Comment (willkunz13) [reply app/views/search/show.html.erb:15]: me too

File: db/migrate/20200412211316_create_garden_plants.rb (hunk 2 above)
Comment (nkeller1) [reply db/migrate/20200412211316_create_garden_plants.rb:3]: yes, but maybe a separate migration when we are ready

**Pull request merged**